### PR TITLE
CNNETANTWO-68  Search

### PR DESCRIPTION
**Pull request by zoitec**
You should be able to pull this branch down and run in local mode and then go to 
http://localhost:5060 and test out the search route in swaggers:

image

### CODE REVIEW

**Comment by zoitec** [comment]
This branch includes changes from Alex's branch 67 which is in PR too.  This one should be reviewed after his one has been approved and merged and after I've merged any updates after his merge.

**Comment by dsover** [comment]
@britneydossett I think so. It is created so that we when deploying it will always have the same dependency trees, incase some of the things we install don't have locked versions. Here is the documentation from npm on it https://github.com/npm/npm/blob/v5.0.0/doc/files/package-lock.json.md 

@jamsyoung do you have any thoughts?

File: src/data-util/search/index.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+const request = require('request');
+
+module.exports = class Content {
+    constructor(config = {}) {
+        this.apiEndpoint = config.apiEndpoint || 'https://search.api.cnn.io';
```
Comment (dsover) [inline src/data-util/search/index.js]: I think we may want to add this to the conf file and maybe a env var. this way if the end point ever changes we don't have to deploy, or if we need different endpoints for different environments

File: src/data-util/search/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            if (config) {
+                self.type = config.type || 'article';
+                self.query = config.query;
+                self.from = config.from;
+                self.size = config.size;
+                self.category = config.category;
+            }
+
+            self.uri = `${self.apiEndpoint}/content?q=${self.query}${self.size ? `&size=${self.size}` : ''}${self.type ? `&type=${self.type}` : ''}${self.category ? `&category=${self.category}` : ''}${self.from ? `&from=${self.from}` : ''}`;
+            console.log('url: ', self.uri);
```
Comment (dsover) [inline src/data-util/search/index.js]: should be using debug

File: src/routes/search/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            config: {
+                description: 'article search',
+                tags: ['api', 'get'],
+                validate: {
+                    query: {
+                        q: server.joi.string()
+                            .required()
+                            .description('The search string.')
+                            .example('world cup'),
+                        size: server.joi.string()
```
Comment (dsover) [inline src/routes/search/index.js]: Should we place a maximum number of search records allowed as part of the validation? this way we can protect the server from overload? or is there a maximum number of results that the search api will return?  

Not required

Comment (zoitec) [inline src/routes/search/index.js]: actually, it seems that the search api sends a max of 10 records at a time anyway by default:

https://search.api.cnn.io/content?q=soccer

if you want more you have to specify that with your size like this like this :
 
https://search.api.cnn.io/content?q=soccer&size=20

**Pull request merged**